core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts

```diff
@@ -759,6 +771,49 @@ export class BaseUser {
     this.page = newPage;
     return newPage;
   }
+
+  /**
+   * This function sets up video recording for the current test.
+   */
+  async setupVideoRecording(): Promise<void> {
+    const outputFileName =
+      `${this.specName}-${new Date().toISOString()}.mp4`.replace(
+        /[^a-z0-9.-]/gi,
+        '_'
+      );
+
+    const outputDir = testConstants.TEST_VIDEO_DIR;
+    if (!fs.existsSync(outputDir)) {
+      fs.mkdirSync(outputDir, {recursive: true});
+    }
+
+    // Skip if there is no page object.
+    if (!this.page) {
+      return;
+    }
+
+    const config = {
+      followNewTab: true,
+      fps: 25,
+      ffmpeg_Path: null,
+      videoFrame: {
+        width: this.page.viewport()?.width || 1920,
+        height: this.page.viewport()?.height || 1080,
+      },
+      aspectRatio: '16:9',
```

Comment:
What would happen if this conflicts with the aspect ratio of the videoFrame dimensions?

Reply:
Removing `aspectRatio` config. Thus, ffmpeg will automatically use aspect ratio of screen.